src/ILLink.Tasks/LinkTask.cs

```diff
@@ -51,11 +51,24 @@ public class ILLink : ToolTask
 
 		/// <summary>
 		///   The directory in which to place linked assemblies.
-		//    Maps to '-out'.
+		///    Maps to '-out'.
 		/// </summary>
 		[Required]
 		public ITaskItem OutputDirectory { get; set; }
 
+		/// <summary>
+		/// Turns off trim correctness analysis warnings.
+		/// Maps to '--nowarn 2006;2026;2037;2043'.
+		/// </summary>
+		public bool TrimmerAnalysisWarnings { set => _trimmerAnalysisWarnings = value; }
```

Comment:
Regardless of where this should live the name needs to change.
Currently we have "Trim correctness" analysis. In the future I anticipate "Single-file correctness", "AOT correctness" and potentially more. The name must reflect this.

I would probably call it `EnableTrimCorrectnessAnalysis` or `TrimCorrectnessAnalysis` (whichever fits the current style in SDK better). It would be "false" by default for now.

Reply:
My understanding is that this should live in the SDK only for now, not in the task. For naming, I personally find `correctness` a little redundant. What do you think of `TrimAnalysis`?

Maybe it would make sense to preemptively open an SDK PR that introduces the option to discuss naming there, and avoid getting caught up in it here? I'm happy to do that if you think it makes sense @vitek-karas @mateoatr

Reply:
> Maybe it would make sense to preemptively open an SDK PR that introduces the option to discuss naming there, and avoid getting caught up in it here? I'm happy to do that if you think it makes sense @vitek-karas @mateoatr

This sounds good. I'm dropping this property from here then. I think `EnableTrimAnalysis` would be a good name, but we can follow up on this conversation in the SDK.

Reply:
@sbomer Please prepare the SDK change (you can do that fastest)... `EnableTrimAnalysis` sounds good - please include Sam in the discussion on the SDK side.

Reply:
https://github.com/dotnet/sdk/pull/12388 to get the discussion started. I will add tests and such as we go.